packages/zoe/src/contracts/vpool-xyk-amm/pool.js

```diff
@@ -0,0 +1,255 @@
+            liquidityValueIn,
+          ),
+          centralBrand,
+        );
+
+        const tokenKeywordAmountOut = AmountMath.make(
+          calcValueToRemove(
+            liqTokenSupply,
+            pool.getSecondaryAmount().value,
+            liquidityValueIn,
+          ),
+          secondaryBrand,
+        );
+
+        liqTokenSupply -= liquidityValueIn;
+
+        poolSeat.incrementBy(userSeat.decrementBy({ Liquidity: liquidityIn }));
+        userSeat.incrementBy(
+          poolSeat.decrementBy({
+            Central: centralTokenAmountOut,
+            Secondary: tokenKeywordAmountOut,
+          }),
+        );
+        zcf.reallocate(userSeat, poolSeat);
+
+        userSeat.exit();
+        updateState(pool);
+        return 'Liquidity successfully removed.';
+      },
+      getNotifier: () => notifier,
+      updateState: () => updateState(pool),
+      // eslint-disable-next-line no-use-before-define
+      getToCentralPriceAuthority: () => toCentralPriceAuthority,
+      // eslint-disable-next-line no-use-before-define
+      getFromCentralPriceAuthority: () => fromCentralPriceAuthority,
+      // eslint-disable-next-line no-use-before-define
+      getVPool: () => vPool,
+    });
+
+    const vPool = makeSinglePool(zcf, pool, protocolFee, poolFee, protocolSeat);
+
+    const getInputPriceForPA = (amountIn, brandOut) =>
+      vPool.getInputPrice(amountIn, AmountMath.makeEmpty(brandOut));
+    const getOutputPriceForPA = (brandIn, amountout) =>
+      vPool.getInputPrice(AmountMath.makeEmpty(brandIn), amountout);
+
+    const toCentralPriceAuthority = makePriceAuthority(
+      getInputPriceForPA,
+      getOutputPriceForPA,
+      secondaryBrand,
+      centralBrand,
+      timer,
+      zcf,
+      notifier,
+      quoteIssuerKit,
+    );
+    const fromCentralPriceAuthority = makePriceAuthority(
+      getInputPriceForPA,
+      getOutputPriceForPA,
+      centralBrand,
+      secondaryBrand,
+      timer,
+      zcf,
+      notifier,
+      quoteIssuerKit,
+    );
+
+    return pool;
+  };
+
+  /**
+   * Allows users to add new liquidity pools. `secondaryIssuer` and
+   * its keyword must not have been already used
+   *
+   * @param {Issuer} secondaryIssuer
+   * @param {Keyword} keyword - will be used in the
+   * terms.issuers for the contract, but not used otherwise
+   */
+  const addPool = async (secondaryIssuer, keyword) => {
+    const liquidityKeyword = `${keyword}Liquidity`;
+    zcf.assertUniqueKeyword(liquidityKeyword);
+
+    const [secondaryAssetKind, secondaryBrand] = await Promise.all([
+      E(secondaryIssuer).getAssetKind(),
+      E(secondaryIssuer).getBrand(),
+    ]);
+
+    assert(
+      !isSecondary(secondaryBrand),
+      X`issuer ${secondaryIssuer} already has a pool`,
+    );
+    assert(
+      secondaryAssetKind === AssetKind.NAT,
+      X`${keyword} issuer must use NAT math`,
+    );
+
+    // We've checked all the foreseeable exceptions (except
```

Comment:
```suggestion
    // COMMIT POINT
    // We've checked all the foreseeable exceptions (except
```

I see such sentinels used in `zoeService/zoeStorageManager.js` etc. It seems in order here.

Reply:
done